tests/cpp/test.cpp

```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 The Foundry Visionmongers Ltd
+
+// Include all headers to test they are where we expect and can be
+// compiled.
+
+#include <openassetio_mediacreation/openassetio_mediacreation.hpp>
+#include <openassetio_mediacreation/traits/content.hpp>
+#include <openassetio_mediacreation/traits/managementPolicy.hpp>
+#include <openassetio_mediacreation/traits/timeline.hpp>
+#include <openassetio_mediacreation/traits/traits.hpp>
+
+using namespace openassetio_mediacreation;
+
+int main() {
+  auto traits = openassetio::TraitsData::make();
+  auto trait = std::make_shared<traits::managementPolicy::ManagedTrait>(traits);
```

Comment:
[consider][minor] No real need to create a `shared_ptr` of a trait view class.  I suspect that's a pretty uncommon use case.  Doesn't do any harm, but is slightly more complex and perhaps obscures this example.

If you want to keep it, we should really include-what-you-use and add `#include <memory>`.

Reply:
@feltech @elliotcmorris  I haven't got this far on the other PR yet - and I appreciate this belongs there - but what is our policy for making these? Does it extend the `make` pattern with private constructor, or are we going with 'whatever you like' as they're entirely host managed objects?

+1 on just using a stack allocated one if so for reasons of clarity - given we _do_ have fancy rules for other classes, and we know people look at tests for guidance.

Reply:
I hope that situation is constrained to derived classes that you have to supply to some other API function via a shared ptr. I dont think we have anything that has to be a shared ptr otherwise, that doesn't have a private constructor.